src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -575,6 +594,84 @@ public function duplicateOrderCartAction(int $orderId)
         );
     }
 
+    /**
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_orders_view",
+     *     redirectQueryParamsToKeep={"orderId"},
+     *     message="You do not have permission to edit this."
+     * )
+     * @DemoRestricted(
+     *     redirectRoute="admin_orders_view",
+     *     redirectQueryParamsToKeep={"orderId"}
+     * )
+     *
+     * @param Request $request
+     * @param int $orderId
+     *
+     * @return Response
+     */
+    public function sendMessageAction(Request $request, int $orderId): Response
+    {
+        $orderMessageForm = $this->createForm(OrderMessageType::class);
+
+        $orderMessageForm->handleRequest($request);
+
+        if ($orderMessageForm->isSubmitted() && $orderMessageForm->isValid()) {
+            $data = $orderMessageForm->getData();
+
+            try {
+                $this->getCommandBus()->handle(new AddOrderCustomerMessageCommand(
+                    $orderId,
+                    $data['message'],
+                    !$data['is_displayed_to_customer']
+                ));
+
+                $this->addFlash(
+                    'success',
+                    $this->trans('Comment successfully added.', 'Admin.Notifications.Success')
+                );
+            } catch (CannotSendEmailException $exception) {
+                $this->addFlash(
+                    'success',
+                    $this->trans('Comment successfully added.', 'Admin.Notifications.Success')
+                );
+
+                $this->addFlash(
+                    'error',
+                    $this->trans(
+                        'An error occurred while sending an email to the customer.',
+                        'Admin.Orderscustomers.Notification'
+                    )
+                );
+            } catch (Exception $exception) {
+                $this->addFlash(
+                    'error',
+                    $this->getErrorMessageForException($exception, $this->getCustomerMessageErrorMapping($exception))
+                );
+            }
+        }
+
+        $routesCollection = $this->get('router')->getRouteCollection();
+
+        if (
```

Comment:
It's a bit hard to understand this `if (...)` statement, what's going on in there ?